auth/principal.go

```diff
@@ -123,6 +123,8 @@ type User interface {
 
 	RoleHistory() TimedSetHistory
 
+	PreloadRoles()
```

Comment:
Is there a reason to not just add GetRoles to the user interface?

Reply:
As discussed, changing to `InitializeRoles()`